samples/secure_services/src/main.c

```diff
@@ -131,15 +166,27 @@ void main(void)
 	}
 
 #ifdef PM_S1_ADDRESS
-	bool s0_active;
+#if CONFIG_SPM_SERVICE_S0_ACTIVE
+	bool s0_active = false;
 
 	ret = spm_s0_active(PM_S0_ADDRESS, PM_S1_ADDRESS, &s0_active);
 	if (ret != 0) {
 		printk("Unexpected failure from spm_s0_active: %d\n", ret);
 	}
 
 	printk("S0 active? %s\n", s0_active ? "True" : "False");
+#elif CONFIG_BUILD_WITH_TFM
+	bool s0_active = false;
 
+	ret = tfm_platform_s0_active(PM_S0_ADDRESS, PM_S1_ADDRESS, &s0_active);
+	if (ret != 0) {
+		printk("Unexpected failure from spm_s0_active: %d\n", ret);
+	}
+
+	printk("S0 active? %s\n", s0_active ? "True" : "False");
+#else
+	printk("No secure service to check if S0 is active\n\n");
+#endif
```

Comment:
Can't we reduce the duplication by half here by moving the checks a bit "further in"? Just around the call to s0_active?